ReduxSimple/ReduxStoreWithHistory.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ReduxSimple
+{
+    public abstract class ReduxStoreWithHistory<TState> : ReduxStore<TState>
+        where TState : class, new()
+    {
+        private readonly Stack<TState> _previousStates = new Stack<TState>();
+        private readonly Stack<TState> _nextStates = new Stack<TState>();
```

Comment:
Same. But we can save both forward `Actions` and `UndoneActions`, or a better name if you have.